docs/tutorials/security/part-2.md

```diff
@@ -0,0 +1,266 @@
+---
+title: 'Part 2: Leaks'
+authors: 'Benjamin Fuentes'
+last_update:
+  date: 11 January 2024
+---
+
+> Note : clone this [project](https://github.com/marigold-dev/training-security-2.git) for compiling and testing this tutorial
+
+## Replay attack
+
+A replay attack on a smart contract is a type of security vulnerability that allows an attacker to reuse a signed valid transaction multiple times. We saw in the previous chapter how to do offchain replay attacks, but it is possible to do also onchain replay attacks.
+Besides, Tezos prevents this kind of vulnerability, it is possible to write a code that does this attack sending the same operation several times for execution.
+
+Compile and simulate the replay attack :
+
+```bash
+taq compile 1-replay.jsligo
+taq simulate 1-replay.tz --param=1-replay.parameter.parameter.tz
+```
+
+The simulation will tell you that several internal transactions will be executed.
+But if you deploy the code and try to execute it :
+
+```bash
+taq deploy 1-replay.tz --mutez 1000 -e testing
+
+taq transfer KT1VMt7t4CboRP6jYBUdBQowHb4NR1UtmDrz -e testing
+```
+
+Then, the Tezos will detect the flaw
+
+```logs
+  "message": "(transaction) proto.017-PtNairob.internal_operation_replay"
+```
+
+## Memory overflow
+
+Memory overflow is a kind of attack that fulfills the memory of a smart contract resulting in making this contract unusable. Even simply loading the data into memory and deserializing it at the beginning of the call, could use so much gas that any call to the contract would fail. All the funds would be forever locked into the contract.
```

Comment:
```suggestion
Memory overflow is a kind of attack that overloads the memory of a smart contract resulting in making this contract unusable. Even simply loading the data into memory and deserializing it at the beginning of the call could use so much gas that any call to the contract would fail. All the funds would be forever locked into the contract.
```